utils/checkdb.py

```diff
@@ -17,7 +17,35 @@
 import glob
 
 
-def make_tile_mask(db_root, tile_name, tilej, strict=False, verbose=False):
+def gen_tile_bits(tile_segbits, tile_bits, strict=False, verbose=False):
+    '''
+    For given tile and corresponding db_file structure yield
+    (absolute address, absolute FDRI bit offset, tag)
+
+    For each tag bit in the corresponding block_type entry, calculate absolute address and bit offsets
+    '''
+
+    for block_type in tile_segbits:
+        assert block_type in tile_bits, "block type %s is not present in current tile" % block_type
+
+        block = tile_bits[block_type]
+
+        baseaddr = block.base_address
+        bitbase = 32 * block.offset
+        frames = block.frames
+
+        for tag in tile_segbits[block_type]:
+            for bit in tile_segbits[block_type][tag]:
+                # 31_06
+                word_column = bit.word_column
+                word_bit = bit.word_bit
+                assert word_column <= frames, "ERROR: bit out of bound --> tag: %s; word_column = %s; frames = %s" % (
+                    tag, word_column, frames)
+                yield word_column + baseaddr, word_bit + bitbase, tag
+
+
+def make_tile_mask(
+        tile_segbits, tile_name, tile_bits, strict=False, verbose=False):
     '''
```

Comment:
These arguments are not used (once gen_tile_bits doesn't take them)?

Reply:
you mean `strict` and `verbose`? If yes, yhey can actually be removed

Reply:
Ya, sorry.  I keep expecting highlighting to show in the comment